neutrinordp/xrdp-neutrinordp.c

```diff
@@ -182,19 +182,33 @@ lxrdp_connect(struct mod *mod)
         }
 
 #endif
-        LOG(LOG_LEVEL_ERROR, "freerdp_connect Failed to "
-            "destination :%s:%d",
+	mod->pid = g_getpid();
+        LOG(LOG_LEVEL_ERROR, "NeutrinoRDP: connection [Failed]," 
+            " destination [%s:%d], username [%s], source [%s:%s],"
+	    " pamusername [%s], pid [%d]",
             mod->inst->settings->hostname,
-            mod->inst->settings->port);
+            mod->inst->settings->port,
+	    mod->inst->settings->username,
+	    mod->client_info.client_addr,
+	    mod->client_info.client_port,
+	    mod->pamusername,
+	    mod->pid);
         return 1;
     }
     else
     {
-        LOG(LOG_LEVEL_INFO, "freerdp_connect returned Success to "
-            "destination :%s:%d",
+	mod->pid = g_getpid();
+        LOG(LOG_LEVEL_INFO, "NeutrinoRDP: connection [Success],"
+            " destination [%s:%d], username [%s], source [%s:%s],"
+	    " pamusername [%s], pid [%d]",
```

Comment:
The message doesn't give much context for the sysadmin. I suggest adding a bit more context to the message like:

```
NeutrinoRDP proxy connection: status [Connected], RDP client [%s:%s], RDP server [%s:%d], RDP server username [%s], proxy process id [%d]
```

Since the pamusername is not used for connecting to the remote RDP server I don't think it makes sense to log it here. Why do you think it should be included in the log message?

Reply:
Thanks! @aquesnel 

Sure. pamusername is not used to connect to a remote RDP server.
I want to record pamusername to keep track of what time and "who" connected to a remote RDP server via NeutrinoRDP Proxy.

Reply:
That makes sense for why to log the pam username here. 

Is it always the case that the pam username is used to authenticate to the proxy and the plain user name is used to authenticate to the remote server? If yes, then if suggest adding the pam username back to the log message with the name "NeutrinoRDP proxy username"

Reply:
The proxy authentication is optional. To my mind `pamusername` is a bad name, but given it's a public interface we're a bit stuck with it.

Reply:
Thanks for the suggestion.

As @matt335672  points out, Proxy PAM authentication is optional.
I assume that the administrator has explicitly set `pamusername=ask` in xrdp.ini.
For this reason, I think it would be easier for the administrator to understand if the log also records `pamusername`.
I suggest the `xrdp pamusername` context.

The log will look like the following
`
[20210505-15:48:12] [INFO ] NeutrinoRDP proxy connection: status [Success], RDP client [192.168.0.212:60050], RDP server [192.168.102.101:3389 ], RDP server username [user], xrdp pamusername [xrdp], xrdp process id [15469].
`

Reply:
That's a good point, and I agree with using `xrdp pamusername`.

I just checked the documentation and it looks like pamusername is an undocumented config option in xrdp.ini.

Reply:
That's true, but there are a few people using it.

The canonical usage is connecting to a VNC server which hasn't been started by sesman (e.g. x11vnc on the console) and using PAM to authenticate the remote connection. That's not relevant here though.

It would be possible to rename the option and add a warning for older usages.

Reply:
The context of the log has been changed in f33a381.